internal/server/http/imposter.go

```diff
@@ -39,15 +39,25 @@ type Response struct {
 	Headers  *map[string]string `json:"headers"`
 }
 
-func findImposters(impostersDirectory string, imposterFileCh chan string) error {
-	err := filepath.Walk(impostersDirectory, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return fmt.Errorf("%w: error finding imposters", err)
-		}
-		if !info.IsDir() && strings.HasSuffix(info.Name(), imposterExtension) {
-			imposterFileCh <- path
-		}
-		return nil
-	})
-	return err
+func findImposters(impostersDirectory string, done <-chan struct{}) (<-chan string, <-chan error) {
+	imposterFiles := make(chan string)
+	errc := make(chan error, 1)
+	go func() {
+		defer close(imposterFiles)
+		errc <- filepath.Walk(impostersDirectory, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return fmt.Errorf("%w: error accsessing path %q", err, path)
+			}
+			if info.IsDir() || strings.HasSuffix(info.Name(), imposterExtension) {
```

Comment:
```suggestion
if info.IsDir() || !strings.HasSuffix(info.Name(), imposterExtension) {
```

be careful, you are ignoring all imposter files: strings.HasSuffix(info.Name(), imposterExtension) you need to negate this condition.

And we will need to check the test because always passing in a way on another, but the test issue is a problem in master also...

Reply:
Thanks for that! I fixed that. Also I changed how `processImposters` handles errors so that we can test that it did fail making it a bit more explicit.